giraffe/src/components/Geo.tsx

```diff
@@ -0,0 +1,237 @@
+  const {lat, lon, zoom, mapStyle, stylingConfig, allowPanAndZoom} = props
+  const {layers, tileServerConfiguration} = props
+  const {tileServerUrl, bingKey} = tileServerConfiguration
+  const mapRef = React.createRef()
+
+  useEffect(() => {
+    ;(mapRef.current as any).leafletElement._onResize()
+  }, [width, height])
+
+  const {table, detectCoordinateFields} = props
+  const [preprocessedTable, setPreprocessedTable] = useState(
+    table
+      ? preprocessData(
+          table,
+          getRowLimit(props.layers),
+          detectCoordinateFields,
+          preprocessedTable => {
+            setPreprocessedTable(preprocessedTable)
+          }
+        )
+      : null
+  )
+
+  useEffect(() => {
+    const newTable = preprocessData(
+      props.table,
+      getRowLimit(props.layers),
+      props.detectCoordinateFields,
+      preprocessedTable => {
+        setPreprocessedTable(preprocessedTable)
+      }
+    )
+    if (newTable) setPreprocessedTable(newTable)
+  }, [table, detectCoordinateFields])
+
+  const onViewportChange = (viewport: {center?: number[]; zoom?: number}) => {
+    const {onViewportChange} = props
+    if (onViewportChange)
+      onViewportChange(viewport.center[0], viewport.center[1], viewport.zoom)
+  }
+
+  return (
+    <Map
+      ref={mapRef}
+      style={{
+        width: `${width}px`,
+        height: `${height}px`,
+      }}
+      center={{lat, lon}}
+      zoom={zoom}
+      minZoom={getMinZoom(width)}
+      zoomDelta={1}
+      zoomSnap={1 / ZOOM_FRACTION}
+      onViewportChanged={onViewportChange}
+      dragging={allowPanAndZoom}
+      zoomControl={allowPanAndZoom}
+      scrollWheelZoom={allowPanAndZoom}
+      attributionControl={false}
+    >
+      {bingKey ? (
+        <LayersControl position="topright">
+          <LayersControl.BaseLayer checked={mapStyle === 'Roads'} name="Roads">
+            <BingLayer minNativeZoom={3} bingkey={bingKey} type="Road" />
+          </LayersControl.BaseLayer>
+          <LayersControl.BaseLayer
+            checked={mapStyle === 'Satellite (plain)'}
+            name="Satellite (Plain)"
+          >
+            <BingLayer minNativeZoom={3} bingkey={bingKey} />
+          </LayersControl.BaseLayer>
+          <LayersControl.BaseLayer
+            checked={mapStyle === 'Satellite'}
+            name="Satellite (Labels)"
+          >
+            <BingLayer
+              minNativeZoom={3}
+              bingkey={bingKey}
+              type="AerialWithLabels"
+            />
+          </LayersControl.BaseLayer>
+          <LayersControl.BaseLayer
+            checked={mapStyle === 'Dark'}
+            name="Dark mode"
+          >
+            <BingLayer minNativeZoom={3} bingkey={bingKey} type="CanvasDark" />
+          </LayersControl.BaseLayer>
+        </LayersControl>
+      ) : (
+        <TileLayer minNativeZoom={3} url={tileServerUrl} />
+      )}
+
+      {layers.map((layer, index) => {
+        if (!preprocessedTable) return
```

Comment:
Please always use braces for if statements. We will be updating the eslint rules to reflect this.